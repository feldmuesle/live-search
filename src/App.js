import React from 'react';
import './App.css';

function App() {
  return <div className="App">Exiting stuff to come...</div>;
}

export default App;
