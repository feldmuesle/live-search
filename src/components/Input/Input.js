import React from 'react';
import PropTypes from 'prop-types';

import './index.css';

function Input({ type, name, onChange, onBlur, value, placeholder, className, isRequired }) {

  const classNames = className ? `input ${className}` : 'input';

  return (
    <input
      type={type}
      name={name}
      onChange={onChange}
      onBlur={onBlur}
      value={value}
      placeholder={placeholder}
      className={classNames}
      required={isRequired}
    />
  );
}

Input.defaultProps = {
  type: 'text',
  placeholder: null,
  className: null,
  isRequired: false,
};

Input.propTypes = {
  type: PropTypes.string,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  className: PropTypes.string,
  isRequired: PropTypes.bool,
};

export default Input;
