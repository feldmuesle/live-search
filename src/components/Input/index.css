.input {
  width: 100%;
  padding: 0.6rem;
  border-radius: 2px;
  border: 1px solid #a8a8a8;
  font-size: 0.9rem;
}

.input:focus {
  box-shadow: 0 0 1px 1px #B5E096;
  border: 1px solid #B5E096;
}
