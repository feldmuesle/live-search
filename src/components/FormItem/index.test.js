import React from 'react';
import renderer from 'react-test-renderer';
import { FormItem } from '.';
import { Input } from '../Input';
import '../../App.css';

describe('FormItem renders correctly', () => {
  it('renders correctly with input', () => {
    const component = renderer.create(
      <FormItem label="Label">
        <Input value="sunshine" name="input" onChange={() => {}} onBlur={() => {}} />
      </FormItem>
    );

    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
