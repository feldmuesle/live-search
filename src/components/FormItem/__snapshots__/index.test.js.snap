// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem renders correctly renders correctly with input 1`] = `
<div
  className="form-item"
>
  <label
    className="form-item__label"
  >
    Label:
    <input
      className="input"
      name="input"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder={null}
      required={false}
      type="text"
      value="sunshine"
    />
  </label>
</div>
`;
